modules/seo-tools/jetpack-seo.php

```diff
@@ -128,16 +128,20 @@ public function meta_tags() {
 			$obj = get_queried_object();
 
 			$meta['description'] = sprintf(
+				/* translators: first property is an user's display name, the second is the site's title. */
 				_x( 'Read all of the posts by %1$s on %2$s', 'Read all of the posts by Author Name on Blog Title', 'jetpack' ),
-				$obj->display_name,
+				( is_object( $obj ) && property_exists( $obj, 'display_name' ) ) ? $obj->display_name : __( 'the author', 'jetpack' ),
 				get_bloginfo( 'title' )
 			);
 		} elseif ( is_tag() || is_category() || is_tax() ) {
-			$obj = get_queried_object();
+			$obj         = get_queried_object();
+			$description = '';
 
-			$description = get_term_field( 'description', $obj->term_id, $obj->taxonomy, 'raw' );
+			if ( is_object( $obj ) && property_exists( $obj, 'term_id' ) && property_exists( $obj, 'taxonomy' ) ) {
+				$description = get_term_field( 'description', $obj->term_id, $obj->taxonomy, 'raw' );
+			}
 
-			if ( ! is_wp_error( $description ) && '' != $description ) {
+			if ( ! is_wp_error( $description ) && ! empty( $description ) ) {
```

Comment:
No need for `empty()`'s error suppression here.

```suggestion
			if ( ! is_wp_error( $description ) && $description ) {
```